From dfa00cb0e1e4320de25a3cb50f51f38e7ea5f8d8 Mon Sep 17 00:00:00 2001
From: Kevin Hao <kexin.hao@windriver.com>
Date: Wed, 7 Sep 2011 09:59:12 +0800
Subject: [PATCH] wrhv/p2020rdb: sync with the DTS of native BSP

Sync with the update of native BSP. The I2C2 and serial2 are
already supported in the past, but missed in the GOS DTS.
So restore them back. The main changes include:
- Add adt7461 support.
- Add I2C bus 2.
- Add serial 2.
- Update the spi property.
- Add gpio.
- Add 1588 support.
- Update crypto property.

Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 arch/powerpc/boot/dts/wrhv_p2020rdb.dts |   67 +++++++++++++++++++++++++++---
 1 files changed, 60 insertions(+), 7 deletions(-)

diff --git a/arch/powerpc/boot/dts/wrhv_p2020rdb.dts b/arch/powerpc/boot/dts/wrhv_p2020rdb.dts
index 64b620b..79aa77c 100644
--- a/arch/powerpc/boot/dts/wrhv_p2020rdb.dts
+++ b/arch/powerpc/boot/dts/wrhv_p2020rdb.dts
@@ -208,6 +208,11 @@
 			interrupt-parent = <&mpic>;
 			dfsrr;
 
+			adt7461@4c{
+				compatible = "adi,adt7461";
+				reg = <0x4c>;
+			};
+
 			eeprom@50 {
 				compatible = "atmel,24c64";
 				reg = <0x50>;
@@ -219,6 +224,22 @@
 			};
 		};
 
+		i2c@3100 {
+			#address-cells = <1>;
+			#size-cells = <0>;
+			cell-index = <1>;
+			compatible = "fsl-i2c";
+			reg = <0x3100 0x100>;
+			interrupts = <43 2>;
+			interrupt-parent = <&mpic>;
+			dfsrr;
+
+			eeprom@52 {
+				compatible = "atmel,24c64";
+				reg = <0x52>;
+			 };
+		};
+
 		serial0: serial@4500 {
 			cell-index = <0>;
 			device_type = "serial";
@@ -229,23 +250,31 @@
 			interrupt-parent = <&mpic>;
 		};
 
+		serial1: serial@4600 {
+			cell-index = <1>;
+			device_type = "serial";
+			compatible = "ns16550";
+			reg = <0x4600 0x100>;
+			clock-frequency = <0>;
+			interrupts = <42 2>;
+			interrupt-parent = <&mpic>;
+		};
+
 		spi@7000 {
 			cell-index = <0>;
 			#address-cells = <1>;
 			#size-cells = <0>;
-			compatible = "fsl,espi";
+			compatible = "fsl,p2020-espi", "fsl,mpc8536-espi";
 			reg = <0x7000 0x1000>;
 			interrupts = <59 0x2>;
 			interrupt-parent = <&mpic>;
-			espi,num-ss-bits = <4>;
-			mode = "cpu";
+			fsl,espi-num-chipselects = <4>;
 
 			fsl_m25p80@0 {
 				#address-cells = <1>;
 				#size-cells = <1>;
-				compatible = "fsl,espi-flash";
+				compatible = "spansion,s25sl12801";
 				reg = <0>;
-				linux,modalias = "fsl_m25p80";
 				spi-max-frequency = <40000000>;
 				mode = <0>;
 
@@ -322,6 +351,15 @@
 			};
 		};
 
+		gpio: gpio-controller@f000 {
+			#gpio-cells = <2>;
+			compatible = "fsl,mpc8572-gpio";
+			reg = <0xf000 0x100>;
+			interrupts = <47 0x2>;
+			interrupt-parent = <&mpic>;
+			gpio-controller;
+		};
+
 		L2: l2-cache-controller@20000 {
 			compatible = "fsl,p2020-l2-cache-controller";
 			reg = <0x20000 0x1000>;
@@ -378,6 +416,16 @@
 			phy_type = "ulpi";
 		};
 
+		ptp_timer: ptimer@24e00 {
+			compatible = "fsl,gianfar-ptp-timer";
+			reg = <0x24e00 0xb0>;
+			interrupts = <68 2 69 2 70 2>;
+			interrupt-parent = <&mpic>;
+			tmr-prsc = <0x2>;
+			cksel = <1>;
+			timer-frequency = <250000000>;
+		};
+
 		enet0: ethernet@24000 {
 			#address-cells = <1>;
 			#size-cells = <1>;
@@ -392,6 +440,7 @@
 			interrupt-parent = <&mpic>;
 			fixed-link = <1 1 1000 0 0>;
 			phy-connection-type = "rgmii-id";
+			ptimer-handle = <&ptp_timer>;
 
 			mdio@520 {
 				#address-cells = <1>;
@@ -427,6 +476,7 @@
 			tbi-handle = <&tbi0>;
 			phy-handle = <&phy0>;
 			phy-connection-type = "sgmii";
+			ptimer-handle = <&ptp_timer>;
 
 			mdio@520 {
 				#address-cells = <1>;
@@ -455,6 +505,7 @@
 			interrupt-parent = <&mpic>;
 			phy-handle = <&phy1>;
 			phy-connection-type = "rgmii-id";
+			ptimer-handle = <&ptp_timer>;
 		};
 
 		sdhci@2e000 {
@@ -469,8 +520,8 @@
 		};
 
 		crypto@30000 {
-			compatible = "fsl,sec3.0", "fsl,sec2.4", "fsl,sec2.2",
-				     "fsl,sec2.1", "fsl,sec2.0";
+			compatible = "fsl,sec3.1", "fsl,sec3.0", "fsl,sec2.4",
+				     "fsl,sec2.2", "fsl,sec2.1", "fsl,sec2.0";
 			reg = <0x30000 0x10000>;
 			interrupts = <45 2 58 2>;
 			interrupt-parent = <&mpic>;
@@ -478,6 +529,8 @@
 			fsl,channel-fifo-len = <24>;
 			fsl,exec-units-mask = <0xbfe>;
 			fsl,descriptor-types-mask = <0x3ab0ebf>;
+			fsl,multi-host-mode = "dual";
+			fsl,channel-remap = <0x3>;
 		};
 
 		mpic: pic@40000 {
-- 
1.7.0.4

